test/unit/GenericPredictorTest.cc

```diff
@@ -82,15 +82,20 @@ TEST_F(GenericPredictorTest, RAS) {
 // correctly, when no address aliasing has occurred
```

Comment:
Do we have a test for explicitly testing the addFTQ function in isolation?

Reply:
Same comment for perceptron

Reply:
We don't because I couldn't think of a way effectively test it.  This is because it only changes the internal state of the BP.  The only way I could think to test it is to do a series of addToFTQ calls, followed by a series of update calls, and then a predict call to check that the BP is in the correct state.  However, this is already being tested in the other tests.

Reply:
That's fair enough